src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -287,61 +314,67 @@ public async Task RemoveAccountAsync(RedirectContext context)
             string currentUri = UriHelper.BuildAbsolute(
                 request.Scheme,
                 request.Host,
                 request.PathBase,
-                microsoftIdentityOptions.CallbackPath.Value ?? string.Empty);
+                _microsoftIdentityOptions.CallbackPath.Value ?? string.Empty);
 
-            if (!applicationOptions.Instance.EndsWith("/"))
-                applicationOptions.Instance += "/";
+            if (!_applicationOptions.Instance.EndsWith("/"))
+                _applicationOptions.Instance += "/";
 
-            string authority ;
-            IConfidentialClientApplication app = null;
+            string authority;
+            IConfidentialClientApplication app;
 
-            if (microsoftIdentityOptions.IsB2C)
+            try
             {
-                authority = $"{applicationOptions.Instance}tfp/{microsoftIdentityOptions.Domain}/{microsoftIdentityOptions.DefaultUserFlow}";
-                app = ConfidentialClientApplicationBuilder
-                    .CreateWithApplicationOptions(applicationOptions)
-                    .WithRedirectUri(currentUri)
-                    .WithB2CAuthority(authority)
-                    .Build();
+                if (_microsoftIdentityOptions.IsB2C)
+                {
+                    authority = $"{ _applicationOptions.Instance}tfp/{_microsoftIdentityOptions.Domain}/{_microsoftIdentityOptions.DefaultUserFlow}";
+                    app = ConfidentialClientApplicationBuilder
+                        .CreateWithApplicationOptions(_applicationOptions)
+                        .WithRedirectUri(currentUri)
+                        .WithB2CAuthority(authority)
+                        .Build();
+                }
+                else
+                {
+                    authority = $"{ _applicationOptions.Instance}{_applicationOptions.TenantId}/";
+
+                    app = ConfidentialClientApplicationBuilder
+                        .CreateWithApplicationOptions(_applicationOptions)
+                        .WithRedirectUri(currentUri)
+                        .WithAuthority(authority)
+                        .Build();
+                }
+
+                // Initialize token cache providers
+                await _tokenCacheProvider.InitializeAsync(app.AppTokenCache).ConfigureAwait(false);
+                await _tokenCacheProvider.InitializeAsync(app.UserTokenCache).ConfigureAwait(false);
+                return app;
             }
-            else
+            catch (Exception ex)
             {
-                authority = $"{applicationOptions.Instance}{applicationOptions.TenantId}/";
-                app = ConfidentialClientApplicationBuilder
-                    .CreateWithApplicationOptions(applicationOptions)
-                    .WithRedirectUri(currentUri)
-                    .WithAuthority(authority)
-                    .Build();
+                _logger.LogInformation(ex.Message);
```

Comment:
_logger has a LogError with an exception argument. Seems more appropriate.

Reply:
fixed. thanks.

---
In reply to: [386016331](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r386016331) [](ancestors = 386016331)